autopilot/contractor.go

```diff
@@ -281,40 +272,31 @@ func (c *contractor) performContractMaintenance(ctx context.Context, w Worker) (
 		c.logger.Warn("could not calculate min score, no hosts found")
 	}
 
-	// fetch consensus state
-	cs, err := c.ap.bus.ConsensusState(ctx)
+	// run host checks
+	checks, err := c.runHostChecks(ctx, hosts, hostData, minScore)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("failed to run host checks, err: %v", err)
 	}
 
-	// create gouging checker
-	gc := worker.NewGougingChecker(state.gs, cs, state.fee, state.cfg.Contracts.Period, state.cfg.Contracts.RenewWindow)
-
-	// prepare hosts for cache
-	hostInfos := make(map[types.PublicKey]hostInfo)
-	for _, h := range hosts {
-		// ignore the pricetable's HostBlockHeight by setting it to our own blockheight
-		h.PriceTable.HostBlockHeight = cs.BlockHeight
-		isUsable, unusableResult := isUsableHost(state.cfg, state.rs, gc, h, minScore, hostData[h.PublicKey])
-		hostInfos[h.PublicKey] = hostInfo{
-			Usable:         isUsable,
-			UnusableResult: unusableResult,
-		}
+	// fetch consensus state
+	cs, err := c.ap.bus.ConsensusState(ctx)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch consensus state, err: %v", err)
 	}
 
-	// update cache.
-	c.mu.Lock()
-	c.cachedHostInfo = hostInfos
-	c.cachedDataStored = hostData
-	c.cachedMinScore = minScore
-	c.mu.Unlock()
-
-	// run checks
-	updatedSet, toArchive, toStopUsing, toRefresh, toRenew, err := c.runContractChecks(ctx, w, contracts, isInCurrentSet, minScore)
+	// run contract checks
+	updatedSet, toArchive, toStopUsing, toRefresh, toRenew := c.runContractChecks(ctx, contracts, isInCurrentSet, checks, cs.BlockHeight)
```

Comment:
I wasn't too happy about passing the host checks here but `IsRedundantIP` is a little annoying. We can either update it twice or pass them, annotate and then pass them to the `bus`. I was on the fence but I figured it's probably fine because `runContractChecks` no longer returns an error, ensuring the host checks will get added to the bus, while at the same time avoiding updating checks twice.

Reply:
This is probably fine. A lot of that will change in the refactor anyway.